bittensor/_neuron/text/advanced_server/nucleus_impl.py

```diff
@@ -87,7 +87,56 @@ def __init__(self,
         
         # -- keeps track of gradients applied
         self.backward_gradients = 0 
-        
+        self.set_fine_tuning_params()
+
+    def set_fine_tuning_params(self):
+        r''' Set to tune only the parameter of the last layer
+        '''
+        def find_last_layer(model):    
+            r''' Recursively find the last layer in a nn.ModuleList
+            '''
+            reverted_child_list = [(name, child) for name, child in model.named_children()]
+            reverted_child_list.reverse()
+
+            for name, child in reverted_child_list:    
+                if isinstance(child, nn.ModuleList):
+                    if self.config.neuron.finetune.num_layers > len(child):
```

Comment:
these could be inlined to avoid too much indentation. In python if you have `if x and y`, if `x` is falsey, then the interpreter actually won't even bother to check for condition `y`. Which is good in this case because it looks like you split it among two lines to avoid an error calling `len` on something that's not a `ModuleList`

Reply:
but the 2 if returns different things
now it is like `if a then (if b then return c) return d`, where both a and b was checked once only.
if we change it to inline then we might have to do it like `if (a and b) then return c elif a then return d` where we might have to check a for 2 times. 
or maybe there is some other better way